### ISSUE TITLE
ClosingParenthesisIndentation reports false positive if first parameter is a hash

### ISSUE THREAD

**Issue by axsuul** [rubocop/rubocop#13909]
# Issue: Incorrect autocorrection by Layout/ClosingParenthesisIndentation cop

## Expected behavior

The `Layout/ClosingParenthesisIndentation` cop should not report an error with the indentation and autocorrect

```ruby
external_service.build_resource({},
  custom_merge: {
    "inventory_level_resources" => [inventory_level_resource],
  },
  version: "inventory_levels",
)
```

## Actual behavior

The `Layout/ClosingParenthesisIndentation` cop reports an error with the indentation of the closing parenthesis in the following Ruby code:

```ruby
external_service.build_resource({},
  custom_merge: {
    "inventory_level_resources" => [inventory_level_resource],
  },
  version: "inventory_levels",
)
```

When attempting to autocorrect, RuboCop produces the following incorrect output:

```ruby
external_service.build_resource({},
  custom_merge: {
    "inventory_level_resources" => [inventory_level_resource],
  },
  version: "inventory_levels",
                               )
```

The closing parenthesis is incorrectly aligned far to the right, which does not match the expected style.

## Steps to reproduce the problem

1. Create a Ruby file with the following content:

   ```ruby
   external_service.build_resource({},
     custom_merge: {
       "inventory_level_resources" => [inventory_level_resource],
     },
     version: "inventory_levels",
   )
   ```

2. Run RuboCop on this file with the command:

   ```bash
   rubocop --auto-correct
   ```

3. Observe the incorrect autocorrection of the closing parenthesis indentation.

## RuboCop version

```
$ rubocop -V
1.71.0
```
